test/integration/integration_test.h

```diff
@@ -10,4 +10,17 @@ class IntegrationTest : public TestBaseWithParam<Network::Address::IpVersion>,
 public:
   IntegrationTest() : HttpIntegrationTest(Http::CodecClient::Type::HTTP1, GetParam(), realTime()) {}
 };
+
+class UpstreamEndpointIntegrationTest : public TestBaseWithParam<Network::Address::IpVersion>,
+                                        public HttpIntegrationTest {
+public:
+  UpstreamEndpointIntegrationTest()
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP1,
+                            /*upstream_address_fn=*/
+                            [](int) {
+                              return Network::Utility::parseInternetAddress(
```

Comment:
Isn't this just duplicating what the default constructor already does? I think you can just remove this and the associated test since it's just testing test code IMO?

Reply:
This test was intended to verify that the new ctor I added that takes an address function actual uses that address for constructing an upstream endpoint. 

If we don't typically add tests for test code, then I am happy to remove as well.

Reply:
In general we don't add tests for test code (though sometimes it happens), but isn't this the same as the default case? So how would you know if it was doing something different? I would probably just remove, but up to you.

Reply:
I changed the test to use the loopback address instead. My argument in favor is so that we don't have untested or unused code (even if it is itself test code), since otherwise this code may never be tested in envoy (only in downstream projects using envoy).

Reply:
SGTM